lib/supavisor.ex

```diff
@@ -160,6 +161,37 @@ defmodule Supavisor do
     end)
   end
 
+  @spec del_all_cache(String.t()) :: [map()]
+  def del_all_cache(tenant) do
+    Logger.info("Deleting all cache for tenant #{tenant}")
+
+    del = fn key, acc ->
+      result = Cachex.del(Supavisor.Cache, key)
+      [%{inspect(key) => inspect(result)} | acc]
+    end
+
+    Supavisor.Cache
+    |> Cachex.stream!()
+    |> Enum.reduce([], fn entry(key: key), acc ->
+      case key do
+        {:metrics, ^tenant} -> del.(key, acc)
+        {:secrets, ^tenant, _} -> del.(key, acc)
+        {:user_cache, _, _, ^tenant, _} -> del.(key, acc)
+        {:tenant_cache, ^tenant, _} -> del.(key, acc)
+        _ -> acc
+      end
+    end)
+  end
+
+  @spec del_all_cache_dist(String.t(), pos_integer()) :: [map()]
+  def del_all_cache_dist(tenant, timeout \\ 15_000) do
+    Logger.info("Deleting all dist cache for tenant #{tenant}")
+
+    for node <- [node() | Node.list()] do
+      %{inspect(node) => :erpc.call(node, Supavisor, :del_all_cache, [tenant], timeout)}
```

Comment:
I chose inspect() because it feels "safe" :), but yeah, I will replace it with to_string (even though it's possible to use just the node name as a key since it's an atom)